packages/build/src/plugins_core/functions/index.js

```diff
@@ -16,23 +17,47 @@ const {
 
 const { getZipError } = require('./error')
 
-const zipFunctionsAndLogResults = async ({ featureFlags, functionsDist, functionsSrc, logs }) => {
-  const isEsbuildEnabled = featureFlags.buildbot_esbuild
+const normalizeFunctionConfig = (functionConfig) => {
+  // The function configuration keys returned by @netlify/config are not an
+  // exact match to the properties that @netlify/zip-it-and-ship-it expects.
+  // We do that translation according to this map.
+  const zisiConfigMap = {
+    external_node_modules: 'externalNodeModules',
+    ignored_node_modules: 'ignoredNodeModules',
+    node_bundler: 'nodeBundler',
+  }
+  const normalizedConfig = mapObject(functionConfig, (key, value) => {
+    const normalizedKey = zisiConfigMap[key] || key
+
+    // When the user selects esbuild as the Node bundler, we still want to use
+    // the legacy ZISI bundler as a fallback. Rather than asking the user to
+    // make this decision, we abstract that complexity away by injecting the
+    // fallback behavior ourselves. We do this by transforming the value
+    // `esbuild` into `esbuild_zisi`, which zip-it-and-ship-it understands.
+    if (normalizedKey === 'nodeBundler' && value === 'esbuild') {
+      return [normalizedKey, 'esbuild_zisi']
+    }
+
+    return [normalizedKey, value]
+  })
+
+  return normalizedConfig
```

Comment:
Would the following be simpler?

```js
const normalizeFunctionConfig = (functionConfig) => {
  return {
    externalNodeModules: functionConfig.external_node_modules,
    ignoredNodeModules: functionConfig.ignored_node_modules,
    nodeBundler: functionConfig.node_bundler === 'esbuild' ? 'esbuild_zisi' : functionConfig.node_bundler,
  }
}
```

Reply:
Yes, definitely! Thank you. 🙌🏻

Reply:
Done in https://github.com/netlify/build/pull/2414/commits/68ea13648dee061d1db9b7bbeb53d1ac7ad99f99.